flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableCompactSinkTest.java

```diff
@@ -0,0 +1,195 @@
+        catalog.close();
+    }
+
+    @BeforeEach
+    void resetTableConf() {
+        tableConf = new Configuration();
+        tableConf.set(FactoryUtil.CONNECTOR, SqlCreateHiveTable.IDENTIFIER);
+        tableConf.set(FileSystemConnectorOptions.AUTO_COMPACTION, true);
+    }
+
+    /** If only sink parallelism is set, compact operator should follow this setting. */
+    @Test
+    void testOnlySetSinkParallelism() throws Exception {
+        final int sinkParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.SINK_PARALLELISM, sinkParallelism);
+
+        assertSinkAndCompactOperatorParallelism(
+                tableConf, true, true, sinkParallelism, sinkParallelism);
+    }
+
+    @Test
+    void testOnlySetCompactParallelism() throws Exception {
+        final int compactParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.COMPACTION_PARALLELISM, compactParallelism);
+
+        assertSinkAndCompactOperatorParallelism(tableConf, false, true, -1, compactParallelism);
+    }
+
+    @Test
+    void testSetBothSinkAndCompactParallelism() throws Exception {
+        final int sinkParallelism = 8;
+        final int compactParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.SINK_PARALLELISM, sinkParallelism);
+        tableConf.set(FileSystemConnectorOptions.COMPACTION_PARALLELISM, compactParallelism);
+
+        assertSinkAndCompactOperatorParallelism(
+                tableConf, true, true, sinkParallelism, compactParallelism);
+    }
+
+    @Test
+    void testSinkAndCompactAllNotSetParallelism() throws Exception {
+        assertSinkAndCompactOperatorParallelism(tableConf, false, false, -1, -1);
+    }
+
+    private void assertSinkAndCompactOperatorParallelism(
+            Configuration tableConf,
+            boolean isSinkParallelismConfigured,
+            boolean isCompactParallelismConfigured,
+            int expectedSinkParallelism,
+            int expectedCompactParallelism)
+            throws Exception {
+        StreamGraph streamGraph = createAndGetStreamGraphContainsSink(tableConf.toMap());
+        assertThat(streamGraph.getStreamNodes()).hasSize(6);
+        StreamNode hiveSinkNode =
+                getFirstStreamNodeByName(streamGraph, HiveTableSink.BATCH_COMPACT_WRITER_OP_NAME);
+        StreamNode compactNode = getFirstStreamNodeByName(streamGraph, BatchSink.COMPACT_OP_NAME);
+        assertThat(hiveSinkNode.isParallelismConfigured()).isEqualTo(isSinkParallelismConfigured);
+        assertThat(compactNode.isParallelismConfigured()).isEqualTo(isCompactParallelismConfigured);
+        if (isSinkParallelismConfigured) {
+            assertThat(hiveSinkNode.getParallelism()).isEqualTo(expectedSinkParallelism);
+        }
+        if (isCompactParallelismConfigured) {
+            assertThat(compactNode.getParallelism()).isEqualTo(expectedCompactParallelism);
+        }
+    }
+
+    private StreamGraph createAndGetStreamGraphContainsSink(Map<String, String> options)
```

Comment:
IIUC, this method is to construct a stream graph, but it looks hard to understand and complex to me. Is it possible to follow what we done like method `HiveTableSourceITCase#testParallelismSetting`.

Reply:
I modify the test logical refer to `ParallelismSettingTest#generateTransformation`.